src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/exceptions/ApplicationExceptionHandler.java

```diff
@@ -93,4 +99,24 @@
         log.error("exception caught", e);
         return exceptionTranslator.toResponseEntity(e, HttpStatus.FORBIDDEN);
     }
+
+    /**
+     * check for nested SignerNotReachable exception when BeanCreationException is caught
```

Comment:
I think it would be good to explain _why_ we do this. What is the special bond between BeanCreationExceptions and SignerNotReachableExceptions? Otherwise some future developer will be confused.

Reply:
Added better documentation here